sql/core/src/main/scala/org/apache/spark/sql/jdbc/DB2Dialect.scala

```diff
@@ -48,5 +48,7 @@ private object DB2Dialect extends JdbcDialect {
     case _ => None
   }
 
-  override def isCascadingTruncateTable(): Option[Boolean] = Some(false)
+  override def getTruncateQuery(table: String): String = {
+    s"TRUNCATE $table"
```

Comment:
Why override the default here?

Reply:
(And all others that are not Postgres?)

Reply:
The `JdbcDialect` object now no longer provides a default implementation for `getTruncateQuery` (as per @dongjoon-hyun 's suggestion). The reason is that this explicitly forces dialects to implement a truncate operation without side effects. 

One thing I don't like about this solution is the code duplication across dialects. But it does prevent dialects from inheriting default behaviour and (accidentally) truncating with side effects.

Reply:
Gotcha. It is also backward incompatible. Ie if someone implemented his/her own dialect one needs to update it (but removing the isCasc...) does this already so I assume that is ok